src/shared/__stories__/drop-target/components/FileManager.jsx

```diff
@@ -0,0 +1,182 @@
+
+const hoverFadeTime = "0.25s";
+
+const FileModalContainer = styled(ModalContainer)`
+  font-family: sans-serif;
+`;
+
+const FileAppBar = styled(AppBar)`
+  background: ${THEME.clientModal.background};
+`;
+
+const Title = styled(Typography)`
+  /* Use greater specificity than Material to override its default styles */
+  #file-manager-modal & {
+    color: #fff;
+  }
+`;
+
+const Body = styled.div`
+  overflow: auto;
+  padding: 20px;
+`;
+
+const VisualTarget = styled.div`
+  align-items: center;
+  background: #ededed;
+  border-radius: 5px;
+  border: 3px dashed transparent;
+  box-sizing: border-box;
+  display: grid;
+  height: 150px;
+  padding: 20px;
+  transition: border ${hoverFadeTime};
+
+  .dragged-over & {
+    border: 3px dashed #333;
+  }
+`;
+
+const VisualTargetContent = styled.div`
+  align-content: center;
+  align-items: center;
+  display: grid;
+  grid-gap: 20px;
+  justify-items: center;
+
+  &:not(.initial-render) * {
+    animation: ${fadeIn} ${hoverFadeTime};
+  }
+`;
+
+const HiddenInput = styled.input`
+  display: none;
+`;
+
+const FileList = styled.div`
+  margin-top: 20px;
+
+  ul {
+    display: flex;
+    flex-wrap: wrap;
+    list-style: none;
+    margin: 0 -5px;
+    padding: 0;
+  }
+
+  li {
+    animation: ${fadeIn} 0.5s;
+    background: #ededed;
+    border-radius: 5px;
+    border: 1px solid #ccc;
+    display: inline;
+    margin: 5px;
+    padding: 5px;
+  }
+`;
+
+let renderNumber = 0;
+
+export default function FileManager() {
+  React.useEffect(() => {
+    renderNumber += 1;
+  });
+
+  const [numHoveredFiles, setNumHoveredFiles] = React.useState(0);
+  const [filenames, setFilenames] = React.useState([]);
+  const fileInput = React.useRef(null);
+
+  const fileCollator = new Intl.Collator("en-US", {
+    numeric: true,
+    sensitifity: "base"
+  });
+
+  function addFilenames(newFilenames) {
+    setFilenames(
+      filenames
+        .concat(newFilenames.filter(fn => !filenames.includes(fn)))
+        .sort(fileCollator.compare)
+    );
+  }
+
+  function onButtonClick() {
+    fileInput.current.click();
+  }
+
+  function handleButtonFiles(e) {
+    addFilenames(Array.from(e.target.files).map(f => f.name));
+  }
+
+  function onHoverStart(e) {
+    setNumHoveredFiles(e.dataTransfer.items.length);
+  }
+
+  function onHoverEnd() {
+    setNumHoveredFiles(0);
+  }
+
+  function onDrop(e) {
+    setNumHoveredFiles(0);
+    addFilenames(Array.from(e.dataTransfer.files).map(f => f.name));
+  }
+
+  return (
+    <FileModalContainer id="file-manager-modal">
+      <FileAppBar position="static">
+        <Toolbar>
+          <Title variant="title">File Manager</Title>
+        </Toolbar>
+      </FileAppBar>
+      <Body>
+        <DropTarget
+          onHoverStart={onHoverStart}
+          onHoverEnd={onHoverEnd}
+          onDrop={onDrop}
+        >
+          <VisualTarget>
+            <VisualTargetContent
+              className={renderNumber === 1 ? "initial-render" : ""}
```

Comment:
I added this to control styling based on whether or not this is the initial render. Basically, I want the button and drop instructions to fade in when the user "undrags", but I don't want them to fade in for the initial render. See these lines:

https://github.com/iodide-project/iodide/blob/issue-2133-homepage-drag-and-drop/src/shared/__stories__/drop-target/components/FileManager.jsx#L58-L60

This may not be the best way of doing it, though.

Reply:
Going to resolve this for now, but I'm definitely still open to suggestions for a better way of doing this. The implementation is a little different now (it's a boolean rather than a number) but the concept is the same.